tinyiiod.c

```diff
@@ -24,10 +24,20 @@ struct tinyiiod {
 	const struct tinyiiod_ops *ops;
 };
 
+enum {
+	READ_BUFFER,
+	WRITE_BUFFER,
+	BUFFER_NO
+};
+
+static char *iiod_buf[BUFFER_NO];
+
 struct tinyiiod * tinyiiod_create(const char *xml,
 				  const struct tinyiiod_ops *ops)
 {
 	struct tinyiiod *iiod = malloc(sizeof(*iiod));
+	iiod_buf[READ_BUFFER] = (char*)malloc(IIOD_BUFFER_SIZE);
```

Comment:
i would change this to calloc. It initializes the buffers to 0.

Reply:
Maybe the question should be if there is a reason to use calloc() instead of malloc()? Since it is slower, I wouldn't to it if it's not necessary.

Reply:
ack.

Reply:
I think there is no need to cast the pointer returned by malloc.